src/Mvc/Mvc.Core/src/ApplicationParts/ApplicationPartAttribute.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationParts
+{
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
+    public sealed class ApplicationPartAttribute : Attribute
```

Comment:
Naming suggestions.

Reply:
What's the concern with current one (ApplicationPartAttribute)?

Reply:
Nothing in particular. People generally tend to have strong opinions about names

Reply:
Is the idea of this new attribute that we want to do something other than what our existing attribute does? I think it's called `RelatedPartAttribute`?

Reply:
We could re-use it here. We generally don't look for `RelatedAssemblyPart` on assemblies that we found using `RelatedAssemblyPart`. However in this case we would need to do for things we discover off of the entry-assembly:

```C#
// MyApp.dll
[RelatedAssemblyPart("MyApp.Views")]
[RelatedAssemblyPart("Rcl")]
[RelatedAssemblyPart("Identity.UI")]
[RelatedAssemblyPart("Identity.UI.Views")]

// Rcl.dll
[RelatedAssemblyPart("Rcl.Views")]

// Identity.UI.dll
[RelatedAssemblyPart("Identity.UI.Views")]
```

The way this would work is - look for `RelatedAssemblyPart` for entry assembly (`MyApp`) and then look for `RelatedAssemblyPart` on those assemblies, but nothing further than that.

Reply:
I'm happy with the new attribute. I wanted to make sure you thought through why it's needed 🤣